toolkit/tools/internal/safechroot/safechroot.go

```diff
@@ -416,6 +417,10 @@ func (c *Chroot) Close(leaveOnDisk bool) (err error) {
 	activeChrootsMutex.Lock()
 	defer activeChrootsMutex.Unlock()
 
+	// Stops gpg-agent and keyboxd if they are running inside the chroot.
+	// This is to avoid leaving folders like /dev mounted when the chroot folder is forcefully deleted in cleanup.
+	c.stopGPGAgent()
```

Comment:
Let's move it past the `if index < 0` check. It should be OK to try and close the same chroot twice and this call will probably now start failing, if you try to run it inside a chroot with `/dev` already unmounted. Also, I think there's no need to stop the agent for the non-regular builds branch - the chroot is not being torn down in this scenario.

Reply:
Updated!